python/graphscope/analytical/app/java_app.py

```diff
@@ -0,0 +1,392 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+#
+# Copyright 2020 Alibaba Group Holding Limited. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# from coordinator.gscoordinator.io_utils import PipeWatcher
+import hashlib
+import json
+import logging
+import os
+import subprocess
+import sys
+import tempfile
+from glob import glob
+from pathlib import Path
+
+import yaml
+
+from graphscope.analytical.udf.utils import InMemoryZip
+from graphscope.framework.app import AppAssets
+from graphscope.framework.app import AppDAGNode
+from graphscope.framework.app import check_argument
+from graphscope.framework.context import create_context_node
+from graphscope.framework.dag import DAGNode
+from graphscope.framework.dag_utils import bind_app
+from graphscope.framework.errors import InvalidArgumentError
+from graphscope.framework.graph import Graph
+from graphscope.proto import graph_def_pb2
+
+__all__ = ["JavaApp"]
+
+logger = logging.getLogger("graphscope")
+
+# runtime workspace
+try:
+    WORKSPACE = os.environ["GRAPHSCOPE_RUNTIME"]
+except KeyError:
+    WORKSPACE = os.path.join("/", tempfile.gettempprefix(), "gs")
+DEFAULT_GS_CONFIG_FILE = ".gs_conf.yaml"
+
+LLVM4JNI_USER_OUT_DIR_BASE = "user-llvm4jni-output"
+JAVA_CODEGNE_OUTPUT_PREFIX = "gs-ffi"
+GRAPHSCOPE_HOME = os.environ.get("GRAPHSCOPE_HOME", "/opt/graphscope")
+GRAPHSCOPE_JNI_LIB = os.path.join(GRAPHSCOPE_HOME, "lib")
+DEFAULT_JVM_CONFIG_FILE = os.path.join(GRAPHSCOPE_HOME, "conf/jvm_options.yaml")
+GRAPHSCOPE_RUNTIME_JAR = os.path.join(GRAPHSCOPE_HOME, "lib/graphscope-runtime-0.1.jar")
```

Comment:
Note that `JavaApp` is defined in the client side, why we need to resolve those options here, rather than on the server side?

Reply:
Good idea. Now I want to put these stuff into Server side.
Bu different from Python/C++ app, for java app, we need to explicitly get the lib path.(For python/c++ we don't need to know the lib path after app is loaded, since we can found it by app_name, from the object manager). Any suggestions on this?